### PR TITLE
ISSUE-441: apply absolute negative patterns to full path

### LINKED ISSUE #441: Ignore patterns are applied to the absolute path instead of  relative to the working directory
### Environment

  * OS Version: Linux
  * Node.js Version: 20.13.1

### Actual behavior

```
[]
```

### Expected behavior

```
['index.test.js']
```

### Steps to reproduce

  1. Create a path `~/something/`
  2. Put a file `index.test.js` in there
  3. Run the thing

### Code sample

index.ts: 
```js
const fg = require('fast-glob');

const include = [ '**/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}' ]
const globOptions = {
  absolute: true,
  ignore: [
    '**/something/**',
  ],
}

const foo = async () => {
        return await fg(include, globOptions);
}

foo().then(console.log);

```

The problem seems to be that the parameter `absolute` which is documented as an output control parameter (https://www.npmjs.com/package/fast-glob#output-control) is getting in the way of filtering.

I have not found the correct place in the code yet, but I figure that paths are RESOLVED first and then FILTERED, instead of FILTERED relatively to `cwd` first and then RESOLVED to their absolute counterparts.
 
I hope what I am getting to is clear. Maybe this is intentional behaviour, but i found no documented evidence of that anywhere.

### PR DESCRIPTION
**Pull request by mrmlnc**
### What is the purpose of this pull request?

Fixes #441, based on an idea from #445.

### What changes did you make? (Give an overview)

Use only absolute (negated) patterns when matching against absolute paths to skip in filter.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -67,7 +67,7 @@
     "test:e2e:stream": "mocha \"out/**/*.e2e.js\" -s 0 --grep \"\\(stream\\)\"",
     "_build:compile": "npm run clean && npm run compile",
     "build": "npm run _build:compile && npm run lint && npm test",
-    "watch": "npm run _build:compile -- --sourceMap --watch",
+    "watch": "npm run _build:compile -- -- --sourceMap --watch",
```
Comment (webpro) [inline package.json:70]: This change shouldn't happen I guess?

**Review by webpro — APPROVED** [review]
Thanks! :)

File: package.json (hunk 1 above)
Comment (webpro) [reply package.json:70]: Oh nm, TIL about the double-double dashes :)

Comment (mrmlnc) [reply package.json:70]: Unfortunately, this is an issue that was already present in the trunk. I have fixed it in the separated commit.

**Comment by mrmlnc** [comment]
```js
Starting bench:regression:async:deep
Label              import.time (time)  time (time)       memory (memory)      entries (value)  process.time (time)
-----------------  ------------------  ----------------  -------------------  ---------------  -------------------
async current **   10.230ms ±1.810%    83.544ms ±4.436%  16.163 MiB ±3.686%   13792 ±0.000%    133.517ms ±3.298%  
async previous **  23.174ms ±1.867%    99.807ms ±2.693%  13.226 MiB ±18.557%  13792 ±0.000%    157.524ms ±1.846%  

Finished bench:regression:async:deep in 43.7s
```

**Pull request merged**